docs/high-availability.md

```diff
@@ -0,0 +1,177 @@
+for ```${NODE_IP}``` on each machine.
+
+
+### Validating your cluster
+Once you copy this into all three nodes, you should have a clustered etcd set up.  You can validate with
+```
+etcdctl member list
+```
+
+and
+
+```
+etcdctl cluster-health
+```
+
+You can also validate that this is working with ```etcdctl set foo bar``` on one node, and ```etcd get foo```
+on a different node.
+
+## Replicated API Servers
+Once you have replicated etcd set up correctly, we will also install the apiserver using the kubelet.
+
+### Installing configuration files
+First you need to create the initial log file, so that Docker mounts a file instead of a directory:
+
+```
+touch /var/log/kube-apiserver.log
+```
+
+Next, you need to create a ```/srv/kubernetes/``` directory on each node.  This directory includes:
+   * basic_auth.csv  - basic auth user and password
+   * ca.crt - Certificate Authority cert
+   * known_tokens.csv - tokens that entities (e.g. the kubelet) can use to talk to the apiserver
+   * kubecfg.crt - Client certificate, public key
+   * kubecfg.key - Client certificate, private key
+   * server.cert - Server certificate, public key
+   * server.key - Server certificate, private key
+
+The easiest way to create this directory, may be to copy it from the master node of a working cluster, or you can manually generate these files yourself.
+
+### Starting the API Server
+Once these files exist, copy the [kube-apiserver.manifest](high-availability/kube-apiserver.manifest) into ```/etc/kubernetes/manifests/``` on each master node.
+
+The kubelet monitors this directory, and will automatically create an instance of the ```kube-apiserver``` container using the pod definition specified
+in the file.
+
+### Load balancing.
+At this point, you should have 3 apiservers all working correctly.  If you set up a network load balancer, you should
+be able to access your cluster via that load balancer, and see traffic balancing between the apiserver instances.  Setting
+up a load balancer will depend on the specifics of your load balancer, for example instructions for the Google Cloud
+Platform can be found [here](https://cloud.google.com/compute/docs/load-balancing/)
+
+Note, if you are using authentication, you may need to regenerate your certificat to include the IP address of the balancer,
+in addition to the IP addresses of the individual nodes.
+
+## Master-elected components
+So far we have set up state storage, and we have set up the API server, but we haven't run anything that actually modifies
+cluster state.  To achieve this reliably, we only want to have one actor modifying state at a time, but we want replicated
+instances of these actors, in case a machine dies.  To achieve this, we are going to use a lease-lock in etcd to perform
+master election.  We'll use a simple utility that runs the master election, and then either creates or deletes a pod
```

Comment:
Maybe mention that in the future, we will use something more tightly integrated, and point to 
 https://github.com/GoogleCloudPlatform/kubernetes/blob/master/docs/proposals/high-availability.md ?